File: 0792-binary-search/0792-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l=0,r=nums.size()-1;
        while(l<=r){
            int i=(l+r)/2;
            if(target<nums[i]){
                r=i-1;
            }
            else if(target > nums[i]){
                l=i+1;

            }
            else{
                return i;
            }
        }
        return -1;
    }
};

File: 0792-binary-search/0792-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l,r=0,len(nums)-1
        while l<=r:
            i=(l+r)//2
            n=nums[i]
            if target<n:
                r=i-1
            elif target>n:
                l=i+1
            else:
                return i

        return -1
        